kfs-core/src/main/java/edu/arizona/kfs/fp/batch/service/impl/ProcurementCardCreateDocumentServiceImpl.java

```diff
@@ -0,0 +1,448 @@
+        DocumentRefreshQueue docRequeue = KewApiServiceLocator.getDocumentRequeuerService("KFS",Long.toString(documentId), 0x0);
+        docRequeue.refreshDocument(Long.toString(documentId));    
+       
+        if ( LOG.isInfoEnabled() ) {
+            LOG.info("Rerouting PCDO document # " + pcardDocumentId + ".");
+        }
+    }
+    
+    /*
+     * as the edu.arizona ProcurementCardDocument implentation has additional fields, we must duplicate this method from the extended class and use the edu.arizona.ProcurementCardDocument implementation(non-Javadoc)
+     * @see org.kuali.kfs.fp.batch.service.impl.ProcurementCardCreateDocumentServiceImpl#createProcurementCardDocuments()
+     */
+    @Override
+    @SuppressWarnings({ "rawtypes", "unchecked" })
+    @Transactional
+    public boolean createProcurementCardDocuments() {
+        List documents = new ArrayList();
+        List cardTransactions = retrieveTransactions();
+
+        // iterate through card transaction list and create documents
+        for (Iterator iter = cardTransactions.iterator(); iter.hasNext();) {
+            documents.add(this.createProcurementCardDocument((List) iter.next()));
+        }
+
+        // now store all the documents
+        for (Iterator iter = documents.iterator(); iter.hasNext();) {
+            ProcurementCardDocument pcardDocument = (ProcurementCardDocument) iter.next();
+            try {
+                documentService.saveDocument(pcardDocument, DocumentSystemSaveEvent.class);
+                if ( LOG.isInfoEnabled() ) {
+                    LOG.info("Saved Procurement Card document: "+pcardDocument.getDocumentNumber());
+                }
+            }
+            catch (Exception e) {
+                LOG.error("Error persisting document # " + pcardDocument.getDocumentHeader().getDocumentNumber() + " " + e.getMessage(), e);
+                throw new RuntimeException("Error persisting document # " + pcardDocument.getDocumentHeader().getDocumentNumber() + " " + e.getMessage(), e);
+            }
+        }
+
+        // send email notification
+        sendEmailNotification(documents);
+
+        return true;
+    }
+    
+    /**
+     * Creates a ProcurementCardDocument from the List of transactions given.
+     *
+     * @param transactions List of ProcurementCardTransaction objects to be used for creating the document.
+     * @return A ProcurementCardDocument populated with the transactions provided.
+     */
+    //copy-pasted from super class to use correct ProcurementCardDocument type (edu.arizona instead of org.kuali)
+    @SuppressWarnings("rawtypes")
+    protected ProcurementCardDocument createProcurementCardDocument( List transactions) {
+        ProcurementCardDocument pcardDocument = null;
+
+        try {
+            // get new document from doc service
+            pcardDocument = (ProcurementCardDocument) SpringContext.getBean(DocumentService.class).getNewDocument(PROCUREMENT_CARD);
```

Comment:
The current standard recommendation is use injection when possible in all these cases...